edb/edgeql/compiler/expr.py

```diff
@@ -121,7 +121,11 @@ def compile_DetachedExpr(
 
 
 @dispatch.compile.register(qlast.Set)
-def compile_Set(expr: qlast.Set, *, ctx: context.ContextLevel) -> irast.Base:
+def compile_Set(
+    expr: qlast.Set,
+    *,
+    ctx: context.ContextLevel
+) -> typing.Union[irast.Set, irast.Expr]:
```

Comment:
After all the changes, I'm not sure that this ever returns anything other than a `irast.Set`. Does it?

Reply:
The bigunion branch calls `dispatch.compile` with this return type.

Reply:
I see.